ww3tools/ProcSat_interpolation.py

```diff
@@ -0,0 +1,515 @@
+        valid_times_unix.append(valid_time_unix.item())
+
+    return model_data_list, np.array(valid_times_unix)
+
+def prepare_and_interpolate(model_data, variable_name, satellite_latitudes, satellite_longitudes, satellite_times, valid_times_unix):
+    mask = spatial_mask(satellite_latitudes, satellite_longitudes, model_data)
+    valid_satellite_points = np.array([satellite_latitudes[mask], satellite_longitudes[mask], satellite_times[mask]]).T
+
+    variable_data = model_data[variable_name]
+    if variable_data.dims != ('latitude', 'longitude', 'time'):
+        variable_data = variable_data.transpose('latitude', 'longitude', 'time')
+
+    model_grid = (variable_data.coords['latitude'].values,
+                  variable_data.coords['longitude'].values,
+                  valid_times_unix)
+
+    interpolator = RegularGridInterpolator(model_grid, variable_data.values, bounds_error=False, fill_value=np.nan)
+    interpolated_values = interpolator(valid_satellite_points)
+
+    # Calculate model times for interpolated values based on satellite times and the model's valid times
+    model_times_for_interpolated_values = np.searchsorted(valid_times_unix, satellite_times[mask], side='right') - 1
+    model_times_for_interpolated_values = valid_times_unix[model_times_for_interpolated_values]
+
+    # Set NaN in model_time_swh and model_time_ws where ws_interpolated is NaN
+    nan_indices = np.isnan(interpolated_values)
+    model_times_for_interpolated_values[nan_indices] = np.nan
+
+    full_interpolated_values = np.full(len(satellite_times), np.nan)
+    full_interpolated_values[mask] = interpolated_values
+
+    full_model_times = np.full(len(satellite_times), np.nan)
+    full_model_times[mask] = model_times_for_interpolated_values
+
+    return full_interpolated_values, full_model_times
+
+def spatial_mask(satellite_latitudes, satellite_longitudes, model_data):
+    min_latitude = model_data.latitude.min().values.item()
+    max_latitude = model_data.latitude.max().values.item()
+    min_longitude = model_data.longitude.min().values.item()
+    max_longitude = model_data.longitude.max().values.item()
+
+    within_lat_bounds = (satellite_latitudes >= min_latitude) & (satellite_latitudes <= max_latitude)
+    within_lon_bounds = (satellite_longitudes >= min_longitude) & (satellite_longitudes <= max_longitude)
+    return within_lat_bounds & within_lon_bounds
+
+
+
+
+# NETCDF interpolator
+def interpolate_netcdf(model_data_directory, model_data_pattern, satellite_file, output_file, model_name):
+    # Using glob to find all files in the directory that match the pattern
+    model_data_files = glob.glob(model_data_directory + model_data_pattern)
+
+    # Check if any files are found
+    if not model_data_files:
+        raise ValueError("No files found in the specified directory matching the pattern.")
+
+    # Sort the files to ensure they are in the correct order
+    model_data_files.sort()
+
+    def extract_reference_times(file_path):
```

Comment:
For this and other def that are only for netcdf would it be helpful to add _netcdf (and _grib to their counterparts) to their names so it's clear one is for netcdf and one if for grib or combine them and have code blocks for each?